---
bool available
float64[] viewpoint